### PR TITLE
JWP scipt

### PR DESCRIPTION
**Pull request by pldespaigne**

### CODE REVIEW

File: apps/backend-ops/src/rescueJWP.ts [hunk 1]
```diff
@@ -0,0 +1,87 @@
+
+import { Movie } from '@blockframes/movie/+state';
+import { loadAdminServices } from '@blockframes/firebase-utils';
+import { MovieVideo } from '@blockframes/movie/+state/movie.firestore';
+import { StorageVideo } from '@blockframes/media/+state/media.firestore';
+
+
+function isVideoOK(video: StorageVideo) {
+  return !!video.jwPlayerId &&
```
Comment (GrandSchtroumpf) [inline apps/backend-ops/src/rescueJWP.ts:9]: can you do : 
```typescript
return ['jwPlayerId', ....].every(key => key in video);
```

File: apps/backend-ops/src/rescueJWP.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  //    - promotional.videos.screener
+
+  const okVideos: MovieVideo[] = [];
+  const emptyVideos: MovieVideo[] = [];
+  const notInJWP: MovieVideo[] = [];
+  const notInStorage: MovieVideo[] = [];
+
+  const sortVideo = (video: StorageVideo, movieId: string) => {
+    if (!video.docId) video.docId = movieId;
+    if (isVideoOK(video)) {
```
Comment (GrandSchtroumpf) [inline apps/backend-ops/src/rescueJWP.ts:47]: For clarity : 
```typescript
if (isVideoOK(video)) return okVideos.push(video);
if (isMissingJW(video)) return notInJWP.push(video);
if (isMissingStorage(video)) return notInStorage.push(video);
return emptyVideos.push(video);
```

File: apps/backend-ops/src/rescueJWP.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+    if (movie.promotional.salesPitch) sortVideo(movie.promotional.salesPitch, movie.id);
+    if (movie.promotional.videos) {
+      sortVideo(movie.promotional.videos.screener, movie.id);
+
+      movie.promotional.videos.otherVideos?.forEach(video => sortVideo(video, movie.id));
+    }
+  });
+
+  console.log('');
```
Comment (GrandSchtroumpf) [inline apps/backend-ops/src/rescueJWP.ts:72]: Do you need this ?

**Review by phpgeekfr — APPROVED** [review]
LGTM, looks harmless

File: apps/backend-ops/src/rescueJWP.ts [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+        notInJWP.push(video);
+      } if (isMissingStorage(video)) {
+        notInStorage.push(video);
+      } else {
+        emptyVideos.push(video);
+      }
+    }
+  }
+
+  const moviesSnap = await db.collection('movies').get();
```
Comment (GrandSchtroumpf) [inline apps/backend-ops/src/rescueJWP.ts:60]: I would do it like that : 
```typescript
const movies = await getCollection<Movie>('movies');
for (const movie of movies) {
  const { salesPitch, videos } = movie.promotional;
  if (salesPitch) {
    sortVideo(salesPitch, movie.id);
  }
  if (videos) {
    sortVideo(videos.screener, movie.id);
    videos.otherVideos?.forEach(video => sortVideo(video, movie.id));
  }
}
```

**Review by GrandSchtroumpf — APPROVED** [review]

**Pull request merged**